### PR TITLE
GCP and Heterogeneous Cloud 

### PR DESCRIPTION
**Pull request by cm891237**
The changes that have been made are to accommodate the necessities of deploying in GCP and having managed clusters in heterogeneous clouds.

This is *not* ready to be merged.

### CODE REVIEW

**Comment by cm891237** [comment]
1. Change group_vars back to blank vars
2. Update README.md with GCP changes
3. Remove commented lines that are there for development

**Comment by cm891237** [comment]
**Sumariner Addon Health Potential Hazard**

With a GCP hub, the `submariner-gateway` pod on the AWS managed clusters in the `sonar-operator` namespace takes a much longer duration to reach a running state than their counterparts in GCP.  The `submariner-gateway` pod needs to be in a running state before the gateways on the submariner addon will be become healthy. 

*Here is an example of the submariner gateway still in a ContainerCreating status after 3mins, i have seen it take over 7*
```
submariner-addon-648557f5b9-fptkm                1/1     Running             0          3m1s
submariner-gateway-cnmxc                         0/1     ContainerCreating   0          13s
```

The GCP managed clusters submariner addons will **sometimes** become healthy in a matter of minutes.  When GCP is very fast and AWS is very slow this creates a condition where the submariner gateways on the GCP managed cluster will never connect find active connections for the AWS managed clusters. See output below. In this condition, the gateways will never become healthy and eventually all the Connection Status's on the ACM console will not be healthy

```
I1209 21:02:12.608306       1 libreswan.go:229] Connection "submariner-cable-aws-us-east-2-10-1-112-157" not found in active connections obtained from whack: map[], map[]
```

We got around this by rolling the submariner-gateway pods on all of the managed clusters, the Submariner-addons become healthy soon after. This has worked every time we have tried.

In our automation we are _only_ estimating the amount of time for the `submariner-gateway` pod on the managed clusters to become healthy, if we underestimate we roll the `submariner-gateway` pods before the pod is ready on the slow managed clusters(s). (still same problem you had before, GKE was probably ready or had given up, AWS still not ready)

If that happens you are back in square 1 , you have to roll them again when all the `submariner-gateway` pods on the managed clusters are RUNNING.

We may want to look at reading into the submariner-addon  to see if there is a status somewhere for the gateway gateway pods. This way we could wait before rolling the gateway pods, thus ensuring they are all rolled at the same time.

**Comment by cm891237** [comment]
*Other Concerns* (Related to ACM)

Reading through ACM docs it is clear that there is room for improvement.

We are using ACM 2.4 (recent migration from 2.3), which has great benefits in terms of speed to detach and more.

According to the [docs](https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.4/html/install/installing) the `Subscription`'s spec uses channel `release-2.3`, which is obviously wrong. 

The real concern is related to the `KlusterletAddonConfig`, which, according to the docs, the `spec.version` changes with no rhyme or reason.
- In ACM [2.4](https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.4/html/clusters/managing-your-clusters) - `.spec.version` = 2.2.0
- In ACM [2.3](https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.3/html/clusters/importing-a-target-managed-cluster-to-the-hub-cluster) - `.spec.version` = 2.3
- in ACM [2.2](https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.2/html/manage_cluster/importing-a-target-managed-cluster-to-the-hub-cluster) - `.spec.version` = 2.2 again

Hard to tell if they will eventually fix this, or if this is how it is intended to be. Needless to say we have to stay wary.

File: group_vars/all.yml [hunk 1]
```diff
@@ -1,42 +1,74 @@
...
+# ex. export OCP_SERVICE_ACCOUNT=$(cat ~/.gcp/osServiceAccount.json)
+
 # Hub context - set to the name of cluster for ACM and hub install
-hub_context: 
+hub_context: hub
 
 # OpenShift Container Platform pull secret retreive from --> cloud.redhat.com/openshift/install/pull-secret
 # add path where pull secret is located 
-ocp_pull_secret_path: 
+ocp_pull_secret_path: /Users/cmwylie19/Public/ansible-multicluster-cockroachdb/pull-secret.json
```
Comment (trujillm) [inline group_vars/all.yml:22]: may want to remove this path

**Review by trujillm — APPROVED** [review]
changes reviews and approved

**Pull request merged**